rekuho/src/lib/Portion.svelte

```diff
@@ -0,0 +1,177 @@
+<!--
+    Portion, may be simply the Editor
+    This window contains the visible and editable text.
+-->
+<script lang="ts">
+  import type { Theme } from "./Theme"
+  import LinesView, { lines, isAlpha, insertAt } from "./LinesView.svelte"
+  import type { Cursor, Position } from "./Cursors.svelte"
+  import CursorsLayer, { cursors, cursorUpdate, cursorMove } from "./Cursors.svelte"
+  import Gutter from "./Gutter.svelte"
+
+  export let theme: Theme
+  const gutter_width = 50 // maybe should be part of theme, minimum value?
+
+  export let height: number
+  export let width: number
+
+  let view: Element
+  let input: HTMLTextAreaElement
+  let container: Element
+
+  // TODO: Get proper input from backend
+  cursors.update((_) => [{ pos: [2, 1] }])
+  lines.update((_) => [
+    ...new Array(10).fill(""),
+    ..."funky\nbanana\nt0wn".split("\n"),
+    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    ...new Array(20).fill("a"),
+  ])
```

Comment:
Not a fan of TODOs in PR's but given the state of the project this is fine

Reply:
Don't agree. Even the rust standard library contains mentions of "TODO" in comments. The specificity of those TODOs will raise over the course of the project.